subprojects/core/src/main/java/org/gradle/internal/progress/BuildOperationService.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.progress;
+
+public interface BuildOperationService {
```

Comment:
Can you mark this up the same way we marked up the build scan request types to indicate that this (and the listener and event types) will be used by build scans and needs to be somewhat stable?

It can be done as a later change after this PR is applied. I don't think it blocks the changes in this PR.